update to use java 17 (#210)
OpenLiberty/guide-liberty-deep-dive: README.adoc

The microservice that you’ll work with is called `inventory`. The `inventory` microservice persists data into a PostgreSQL database. 

image::inventory.png[Inventory microservice,align="center",height=85%,width=85%]

== Additional prerequisites

Docker must be installed before you start the **Persisting Data** module. For installation instructions, refer to the https://docs.docker.com/get-docker/[official Docker documentation^]. You'll build and run the application in Docker containers.

Make sure to start your Docker daemon before you proceed.

Also, if you are using Linux, Kubernetes must be installed before you start the **Deploying the microservice to Kubernetes**.

> You will use `Minikube` as a single-node Kubernetes cluster that runs locally in a virtual machine. Make sure you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^].

Reviewer: ```suggestion
You will use `Minikube` as a single-node Kubernetes cluster that runs locally in a virtual machine. Make sure that you have `kubectl` installed. If you need to install `kubectl`, see the https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux[kubectl installation instructions^].
```

* Under Group specify: `io.openliberty.deepdive`
* Under Artifact specify: `inventory`
* Under Build Tool select: `Maven`
* Under Java SE Version select: `17`
* Under Java EE/Jakarta EE Version select: `10.0`
* Under MicroProfile Version select: `6.0`

ifndef::cloud-hosted[]
Then, click `Generate Project`, which downloads the starter project as `inventory.zip` file. 

> Next, extract the `inventory.zip` file on your system. Move the contents of this extracted `inventory` directory to the `start` directory of this project, which is located at the following path: `guide-liberty-deepdive/start/inventory`.

Reviewer: ```suggestion
Next, extract the `inventory.zip` file on your system. Move the contents of this extracted `inventory` directory to the `start` directory of this project, which is at the following path: `guide-liberty-deepdive/start/inventory`.
```

[INFO] [AUDIT] CWWKF0011I: The server defaultServer is ready to run a smarter planet.
----

> When you need to stop the Liberty instance, press `CTRL+C` in the command-line session where you ran the Liberty.

Reviewer: just to confirm is it `CTRL+C` or is it `CTRL+c` or press `ctrl+c` ?

Author: actually is `c`. Because used `C` across all guides, we will update together if need.

include::finish/module-config/src/main/liberty/config/server.xml[]
----

> Add variables for the [hotspot=httpPortVariable file=0]`HTTP` port, [hotspot=httpsPortVariable file=0]`HTTPS` port, and the [hotspot=contextRootVariable file=0]`context root` to the `server.xml` configuration file. Change the [hotspot=editedHttpEndpoint file=0]`httpEndpoint` element to reflect the new `http.port` and `http.port` variables and change the [hotspot=editedContextRoot file=0]`contextRoot` to use the new `context.root` variable too.

Reviewer: ```suggestion
Add variables for the [hotspot=httpPortVariable file=0]`HTTP` port, [hotspot=httpsPortVariable file=0]`HTTPS` port, and the [hotspot=contextRootVariable file=0]`context root` to the `server.xml` configuration file. Change the [hotspot=editedHttpEndpoint file=0]`httpEndpoint` element to reflect the new `http.port` and `http.port` variable and change the [hotspot=editedContextRoot file=0]`contextRoot` to use the new `context.root` variable too.
```

Author: I updated to
> ... the new `http.port` and `https.port` variables ...

The [hotspot=jwtSsoFeature file=0]`jwtSso` feature adds the libraries that are required for JWT SSO implementation. Configure the [hotspot=jwtSsoConfig file=0]`jwtSso` feature by adding the [hotspot=jwtBuilder file=0]`jwtBuilder` configuration to your `server.xml` file. Also, configure the MicroProfile [hotspot=mpJwt file=0]`JWT` with the `audiences` and `issuer` properties that match the [hotspot file=1]`microprofile-config.properties` defined at the `system/src/main/webapp/META-INF` directory under the `system` project. For more information, see the https://www.openliberty.io/docs/latest/reference/feature/jwtSso-1.0.html[JSON Web Token Single Sign-On feature^], https://www.openliberty.io/docs/latest/reference/config/jwtSso.html[jwtSso element^], and https://www.openliberty.io/docs/latest/reference/config/jwtBuilder.html[jwtBuilder element^] documentation.

> The [hotspot=keyStore file=0]`keyStore` element is used to define the repository of security certificates used for SSL encryption. The `id` attribute is a unique configuration ID that is set to `guideKeyStore`. The `password` attribute is used to load the keystore file, and its value can be stored in clear text or encoded form. To learn more about other attributes, see the https://openliberty.io/docs/latest/reference/config/keyStore.html#keyStore.html[keyStore^] attribute documentation. 

Reviewer: ```suggestion
The [hotspot=keyStore file=0]`keyStore` element is used to define the repository of security certificates used for SSL encryption. The `id` attribute is a unique configuration ID that is set to `guideKeyStore`. The `password` attribute is used to load the keystore file, and its value can be stored in clear text or encoded form. To learn more about other attributes, see the https://openliberty.io/docs/latest/reference/config/keyStore.html#keyStore.html[keyStore attribute documentation^]. 
```

// Starting and preparing your cluster for deployment
ifndef::cloud-hosted[]
== Starting and preparing your cluster for deployment

If you are using Linux, you can continue this section.

Start your Kubernetes cluster.

Run the following command from a command-line session:

[role=command]
```
minikube start
```

> Next, validate that you have a healthy {kube} environment by running the following command from the active command-line session.

Reviewer: Should we be saying " healthy {kube} environment " or " healthy kubernetes environment " ?

Author: I updated "{kube}" to "Kubernetes".

[role=command]
```
kubectl get nodes
```

This command should return a `Ready` status for the `minikube` node.

Run the following command to configure the Docker CLI to use Minikube's Docker daemon.
After you run this command, you will be able to interact with Minikube's Docker daemon and build new
images directly to it from your host machine:

[role=command]
```
eval $(minikube docker-env)
```

Rebuild your Docker image under Minikube with the following commands:

[role='command']
```
docker build -t liberty-deepdive-inventory:1.0-SNAPSHOT .
```
endif::[]

== Deploying the microservice to Kubernetes

ifndef::cloud-hosted[]
If you are using Linux, you can continue this section.
endif::[]

Now that the containerized application is built and tested, deploy it to a local Kubernetes cluster. 

=== Installing the Open Liberty Operator 

ifndef::cloud-hosted[]
Install the Open Liberty Operator to deploy the microservice to Kubernetes.

> First, install the cert-manager to your Kubernetes cluster by running the following command:

Reviewer: ```suggestion
First, install the `cert-manager` to your Kubernetes cluster by running the following command:
```

[role='command']
```
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.3/cert-manager.yaml
```

Next, install Custom Resource Definitions (CRDs) for the Open Liberty Operator by running the following command:
[role='command']
```
kubectl apply --server-side -f https://raw.githubusercontent.com/OpenLiberty/open-liberty-operator/main/deploy/releases/1.2.1/kubectl/openliberty-app-crd.yaml
```
Custom Resources extend the Kubernetes API and enhance its functionality.

> Set environment variables for namespaces for the Operator by running the following commands:

Reviewer: ```suggestion
Set environment variables for namespaces for the operator by running the following commands:
```

Author: I updated to "the Open Liberty Operator"

olo-controller-manager-6fc6b456dc-s29wl 1/1     Running   0          10m
----

> Wait for a while to let the inventory service to start, then run the following command to set up port forwarding to access the `inventory` microservice:

Reviewer: ```suggestion
Pause briefly to give the inventory service time to initialize. After it has started, use the following command to configure port forwarding to access the `inventory` microservice:
```

You can modify the inventory deployment to customize the service. Customizations for a service include changing the port number, changing the context root, and passing confidential information by using Secrets. 

> The [hotspot=contextRoot file=0]`context.root` variable is defined in the [hotspot file=0]`server.xml` configuration file. The context root for the inventory service can be changed by using this variable. The value for the [hotspot=contextRoot file=0]`context.root` variable can be defined in a ConfigMap and accessed as an environment variable.

Reviewer: ```suggestion
The [hotspot=contextRoot file=0]`context.root` variable is defined in the [hotspot file=0]`server.xml` configuration file. The context root for the inventory service can be changed by using this variable. The value for the [hotspot=contextRoot file=0]`context.root` variable can be defined in a `ConfigMap` and accessed as an environment variable.
```

=== Tearing down the environment 

> When you're finished trying out the microservice, press `CTRL+C` in the command line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:

Reviewer: just to confirm is it `CTRL+C` or is it `CTRL+c` or `ctrl+c` ?

Author: actually is c. Because used C across all guides, we will update together if need.

> When you're finished trying out the microservice, press `CTRL+C` in the command line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:

Reviewer: ```suggestion
When you're finished trying out the microservice, press `CTRL+C` in the command-line session where you ran the `kubectl port-forward` command to stop the port forwarding. You can delete all Kubernetes resources by running the `kubectl delete` commands:
```
